polygeist/tools/cgeist/Lib/CGCall.cc

```diff
@@ -328,14 +328,13 @@ ValueCategory MLIRScanner::callHelper(
     for (int I = 0; I < 3; I++) {
       Value Val = L0.val;
       if (auto MT = Val.getType().dyn_cast<MemRefType>()) {
-        Value Idx[] = {getConstantIndex(0), getConstantIndex(I)};
         assert(MT.getShape().size() == 2);
         Blocks[I] = Builder.create<arith::IndexCastOp>(
             Loc, IndexType::get(Builder.getContext()),
-            Builder.create<memref::LoadOp>(Loc, Val, Idx));
+            Builder.create<memref::LoadOp>(
+                Loc, Val,
+                ArrayRef<Value>({getConstantIndex(0), getConstantIndex(I)})));
```

Comment:
Can we use `ValueRange` instead of `ArrayRef<Value>`?

Reply:
I tried in one case (not in this PR) and caused failures (could be some other issue, I didn't pursue) because I looked at what MLIR does and found they do use ArrayRef.

Reply:
That said let me try in isolation to use ValueRange if it works I can change it.

Reply:
ok it works (I must have made some other mistake before), changed.

Reply:
> Minor suggestions. Also, some of the explicit type changes are not needed (`getType()`, `getLocation()`, `getValue()` or common functions like `Visit`)

I applied the changes to use ValueRange rather than ArrayRef<Value>. I kept the other changes in order to follow the LLVM coding guidelines w.r.t. to the use of `auto`. I realize the changes aren't needed of course (this is a NFC PR) but I feel we should follow the LLVM coding guidelines more strictly.